src/query.rs

```diff
@@ -2,7 +2,39 @@
 
 use std::convert::From;
 use linear_map::LinearMap;
-use value::*;
+use value::{Key, Pointer, Value};
+
+/// Specifies a positive numeric range of data.
+pub struct Range(Bound<u64>, Bound<u64>);
```

Comment:
Might as well make it `Range<T>` and then have a type alias where it needs to be a `u64`. Also why not use rust's [`Range`](https://doc.rust-lang.org/std/ops/struct.Range.html)?